internal/resources/network_peer_schema.go

```diff
@@ -0,0 +1,69 @@
+package resources
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/objectplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func NetworkPeerSchema() schema.Schema {
+	return schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"organization_id": stringAttribute([]string{required, requiresReplace}),
+			"project_id":      stringAttribute([]string{required, requiresReplace}),
+			"cluster_id":      stringAttribute([]string{required, requiresReplace}),
+			"name":            stringAttribute([]string{required}),
+			"provider_type":   stringAttribute([]string{required}),
+			"commands": schema.SetAttribute{
+				Computed:    true,
+				ElementType: types.StringType,
+			},
+			"provider_config": schema.SingleNestedAttribute{
+				//	Description: "The 'accountId', 'vpcId', 'region', and 'cidr' fields are required for AWS VPC peering. " +
+				//		"For GCP, the 'networkName', 'projectId', 'serviceAccount', and 'cidr' fields are required for VPC peering. ",
+				Required: true,
+				PlanModifiers: []planmodifier.Object{
+					objectplanmodifier.RequiresReplace(),
+				},
+				Attributes: map[string]schema.Attribute{
+					"aws_config": schema.SingleNestedAttribute{
+						Optional: true,
+						Attributes: map[string]schema.Attribute{
+							"account_id":  stringAttribute([]string{optional}),
+							"vpc_id":      stringAttribute([]string{optional}),
+							"region":      stringAttribute([]string{optional}),
+							"cidr":        stringAttribute([]string{required}),
+							"provider_id": stringAttribute([]string{computed}),
+						},
+					},
+					"gcp_config": schema.SingleNestedAttribute{
+						Optional: true,
+						Attributes: map[string]schema.Attribute{
+							"cidr":            stringAttribute([]string{required}),
+							"network_name":    stringAttribute([]string{optional}),
```

Comment:
No these cannot be required fields, they have to be optional, thats because we need to use either the aws_config or gcp_config as the provider_config and each of these structs have different fields, so none of them can be required otherwise it will ask to input all the fields, note here that cidr is a common field which is required for both, hence it is kept as required, all others have to be optional or else we won't be able to provide either one of the aws or gcp as the configuration.